Update building-and-testing-nodejs examples (#25312)
github/docs: content/actions/automating-builds-and-tests/building-and-testing-nodejs.md

    strategy:
      matrix:

>         node-version: [14.x, 16.x, 18.x, 20.x]

Reviewer: Also removing version 14 based on https://github.com/github/docs/pull/25312#issuecomment-1532802449
```suggestion
        node-version: [16.x, 18.x, 20.x]
```

The `setup-node` action takes a Node.js version as an input and configures that version on the runner. The `setup-node` action finds a specific version of Node.js from the tools cache on each runner and adds the necessary binaries to `PATH`, which persists for the rest of the job. Using the `setup-node` action is the recommended way of using Node.js with {% data variables.product.prodname_actions %} because it ensures consistent behavior across different runners and different versions of Node.js. If you are using a self-hosted runner, you must install Node.js and add it to `PATH`.

> The starter workflow includes a matrix strategy that builds and tests your code with four Node.js versions: 14.x, 16.x, 18.x, and 20.x. The 'x' is a wildcard character that matches the latest minor and patch release available for a version. Each version of Node.js specified in the `node-version` array creates a job that runs the same steps.

Reviewer: ```suggestion
The starter workflow includes a matrix strategy that builds and tests your code with three Node.js versions: 16.x, 18.x, and 20.x. The 'x' is a wildcard character that matches the latest minor and patch release available for a version. Each version of Node.js specified in the `node-version` array creates a job that runs the same steps.
```

Each job can access the value defined in the matrix `node-version` array using the `matrix` context. The `setup-node` action uses the context as the `node-version` input. The `setup-node` action configures each job with a different Node.js version before building and testing code. For more information about matrix strategies and contexts, see "[AUTOTITLE](/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix)" and "[AUTOTITLE](/actions/learn-github-actions/contexts)."

```yaml{:copy}
strategy:
  matrix:

>     node-version: [14.x, 16.x, 18.x, 20.x]

Reviewer: ```suggestion
    node-version: [16.x, 18.x, 20.x]
```

```yaml{:copy}
strategy:
  matrix:

>     node-version: [10.17.0, 17.9.0]

Reviewer: ```suggestion
    node-version: [17.9.0, 18.12.0]
```
